server/visualdl/lib.py

```diff
@@ -41,13 +43,19 @@ def get_scalar(storage, mode, tag, num_records=300):
         if len(data) <= num_records:
             return data
 
-        span = float(len(data) / num_records)
-        end_idx = len(data) - 1
-        res = []
-        for i in xrange(num_records):
-            id = int(end_idx - i * span)
-            res.append(data[id])
-        return [v for v in reversed(res)]
+        span = float(len(data)) / (num_records - 1)
+        span_offset = 0
+
+        data_idx = int(span_offset * span)
+        sampled_data = []
+
+        while data_idx < len(data):
+            sampled_data.append(data[len(data) - data_idx - 1])
```

Comment:
len(...) will run every time in the iter
place it outside the whileloop like:

```
data_size = len(data)

while ...:
    ...
```

Reply:
done